pkg/clusterrouter/proxy/proxy_server.go

```diff
@@ -233,12 +234,18 @@ func (r *RemoteService) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 			er.Error(rw, req, validation.Unauthorized)
 			return
 		}
-		token, err := r.getImpersonatorAccountToken(userInfo)
-		if err != nil && !strings.Contains(err.Error(), dialer2.ErrAgentDisconnected.Error()) {
-			er.Error(rw, req, fmt.Errorf("unable to create impersonator account: %w", err))
-			return
+
+		if !authcontext.IsSAAuthenticated(req.Context()) {
+			// If the request is not authenticated as a service account,
+			// we need to use an impersonation token.
```

Comment:
I'd recommend updating this comment to elaborate more on why impersonation is necessary for users, but not service accounts.
```suggestion
			// Since downstream SA requests are made using the provided SA token, 
			// we don't need to use impersonation for them like we do for Rancher users, 
			// which are using Rancher tokens only found in upstream 	
```